lms/djangoapps/discussion/rest_api/api.py

```diff
@@ -1165,6 +1165,64 @@ def get_response_comments(request, comment_id, page, page_size, requested_fields
         raise CommentNotFoundError("Comment not found") from err
 
 
+def get_user_comments(
+    request: Request,
+    user_id: str,
+    course_id: str,
+    flagged: bool = False,
+    page: int = 1,
+    page_size: int = 10,
+    requested_fields: Optional[List[str]] = None,
+):
+    """
+    Returns the list of comments made by the user in the requested course.
+
+    Arguments:
+
+        request: The django request object.
+
+        user_id: The id of the user to get comments from.
+
+        course_id: The id of the course to filter the comments.
+
+        flagged: Filter comments by flagged status.
+
+        page: The page number (1-indexed) to retrieve
+
+        page_size: The number of comments to retrieve per page
+
+        requested_fields: Indicates which additional fields to return for
+        each child comment. (i.e. ['profile_image'])
+
+    Returns:
+
+        A paginated result containing a list of comments.
+    """
+    try:
+        response = Comment.retrieve_all({
+            'user_id': user_id,
+            'course_id': course_id,
+            'flagged': flagged,
+            'page': page,
+            'page_size': page_size,
+        })
+        if not response["collection"] and page != 1:
+            raise PageNotFoundError("Page not found (No results on this page).")
+
+        context = {}  # TODO How to get this?
```

Comment:
I've looked into this a bit, and I think you can do the following:
1. Just call `get_context` with `thread=None`. This will return enough context for this API but won't include thread-specific data. 
2. Now, some Comment serialisation code expects the thread context, update the code to work without the context.

Reply:
@xitij2000 I made some changes to the PR, implementing the user comments/threads endpoints. Could you take a look if this is looking reasonable so far?